### ISSUE TITLE
RowClicked event & Number of rows to display

### ISSUE THREAD

**Issue by boukenka** [NonFactors/MVC5.Grid#20]
Hello again,

I have 2 new questions:

A. Here is the code to display the grid information:
```C#
@(Html.Grid(Model).Named("PersonGrid")
    .Build(columns =>
    {
        columns.Add(model => Html.CheckBox(model.Id.ToString())).Encoded(false);
        columns.Add(model => model.Id).Css("hidden");
        columns.Add(model => model.ViewName).Titled("Name").Filterable(true).Sortable(true);
        columns.Add(model => model.Surname).Titled("Surname").Filterable(false);
        columns.Add(model => model.ViewAge).Titled("Age").Encoded(false)
            .RenderedAs(model => "<span style='color: #CC4444'>" + model.ViewAge + "</span>");
        columns.Add(model => model.BirthDate).Titled("Birth date").Formatted("{0: yyyymmdd}");
        columns.Add(model => model.IsWorking).Titled("Employed").Css("bolded");
        columns.Add(model => Html.Hidden("")).Titled("Action").Encoded(false)
            .RenderedAs(model => "<a onclick='suppress(" + model.Id + ");return false;'>Remove</a>");
    })
    .ProcessWith(new Research14.Models.ShortNameProcessor())
    .Filterable()
    .Sortable()
    .Empty("No data found")
    .Pageable(pager =>
    {
        pager.InitialPage = 1;
        pager.PagesToDisplay = 10;
        pager.RowsPerPage = 4;
    })
)
```
The script (not all) for the grid is:
```JS
var suppress = function (id) {
    alert(id);
};

var activate = function () {
    $('.mvc-grid').mvcgrid({
        rowClicked: function (grid, row, data) {
            $(row).siblings().removeClass('selected');
            $(row).toggleClass('selected');
            objectId = data['Id'];
            show();
            $.get('/Home/Edit/', { id: objectId })
             .success(function (data) {
                 $("#personContent").html(data);
                 $('#personForm').jQueryUIHelpers();
                 hide();
             })
            .fail(function () {
                hide();
                alert("Error to load person details!");
            })
        }
    })
};

$(function () {
    $('.mvc-grid').mvcgrid();
    activate();
    $('.mvc-grid').mvcgrid({
        reloadStarted: function (grid) {
            show();
        },
        reloadEnded: function (grid) {
            $("#personContent").html('');
            hide();
        },
        reloadFailed: function (grid, result) {
            hide();
        }
    });
});
```
When there is a click on the 'Remove' link in the grid, there is a call to 'suppress' function. Then the rowClicked event is triggered. The same event is also triggered when you click on a checkbox of the grid. Is there a way that in the case, the rowClicked event is not triggered? If the event must be triggered, if there a way to identify what has been clicked e.g. checkbox, 'Remove' link or another value?

B. How you will implement that on the web page, a drop down is available with possible values to select the number of rows e.g. 5, 10, 20 (pager.RowsPerPage) to display in the grid? Could you propose an example please?

Thanks

**Comment by Muchiachio** [comment]
A) You can stop event propagation with jQuery's [event.stopPropagation](https://api.jquery.com/event.stoppropagation/) method.
```JS
$('.mvc-grid-row a').on('click', function(e) {
    e.stopPropagation();
});
```
B) Where is no simple way to do this now, you will need to implement your own [pager](https://github.com/NonFactors/MVC5.Grid/blob/develop/src/Mvc.Grid.Core/Paging/GridPager.cs) class, to handle paging as needed. I may add an example for this in following weeks. However I will add support for this in v3.0, or even in 2.x in the near future.

**Comment by boukenka** [comment]
Thanks for the answers.

For A, I have tried. But it didn't work. I have made some changes. I have done the following for the view (adding the event information):

columns.Add(model => Html.Hidden("")).Titled("Action").Encoded(false).RenderedAs(model => "<a onclick='suppress(event," + model.Id + ");return false;'>Remove</a>");

Then in the script 'suppress', I have added the 'stopPropagation' function.

var suppress = function (event, id) {
    alert(id);
    event.stopPropagation();
};

Now it is working!
